packages/pwa/app/utils/url.js

```diff
@@ -5,9 +5,25 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import {DEFAULT_LOCALE} from '../constants'
-import {getUrlConfig} from './utils'
-import {urlPartPositions} from '../constants'
+import {getAppOrigin} from 'pwa-kit-react-sdk/utils/url'
+import {getParamsFromPath, getUrlConfig} from './utils'
+import {getDefaultSite} from './site-utils'
+import {HOME_HREF, urlPartPositions} from '../constants'
+
+/**
+ * A function that takes a path and qualifies it with the current host and protocol.
+ * This function works on the client and on the server.
+ *
+ * @example
+ * urlToPath(/women/dresses?color=black)
+ *
+ * // returns //http(s)://www.site.com/women/dresses?color=black
+ * @param path
+ * @returns {string|*}
+ */
+export const pathToUrl = (path) => {
+    return `${getAppOrigin()}${path}`
```

Comment:
```suggestion
    return new URL(path, getAppOrigin()).toString()
```
If we use `URL` object, we can be more flexible in the type of `path` we receive:
- if `path` is mistakenly passed in with an _absolute_ url, this implementation would still be able to handle it and simply return path as it is.
- if `path` is correctly a _relative_ url, the implementation still gives you the same result as before.